incubator/hnc/internal/validators/namespace.go

```diff
@@ -137,7 +137,17 @@ func (v *Namespace) cannotDeleteSubnamespace(req *nsRequest) admission.Response
 }
 
 func (v *Namespace) illegalCascadingDeletion(ns *forest.Namespace) admission.Response {
-	if ns.AllowsCascadingDeletion() || len(ns.ChildNames()) == 0 {
+	if ns.AllowsCascadingDeletion() {
+		return allow("")
+	}
+
+	subNs := []string{}
```

Comment:
Do we need to build a list? Or can we just return as soon as we find the first subnamespace?

Reply:
Done